presidio-analyzer/presidio_analyzer/analyzer_engine.py

```diff
@@ -223,6 +227,116 @@ def analyze(
 
         return results
 
+    def analyze_iterator(
+        self,
+        list_of_texts: Iterable[Union[str, bool, float, int]],
+        language: str,
+        **kwargs,
+    ) -> List[List[RecognizerResult]]:
+        """
+        Analyze an iterable of strings.
+
+        :param list_of_texts: An list containing strings to be analyzed.
+        :param language: Input language
+        :param kwargs: Additional parameters for the `AnalyzerEngine.analyze` method.
+        """
+
+        # validate type
+        list_of_texts = (text for text in list_of_texts if self._validate_type(text))
+
+        # Process the texts as batch for improved performance
+        nlp_artifacts_batch: Iterator[
+            Tuple[str, NlpArtifacts]
+        ] = self.nlp_engine.process_batch(texts=list_of_texts, language=language)
+
+        list_results = []
+        for text, nlp_artifacts in nlp_artifacts_batch:
+            results = self.analyze(
+                text=str(text), nlp_artifacts=nlp_artifacts, language=language, **kwargs
+            )
+
+            list_results.append(results)
+
+        return list_results
+
+    @staticmethod
+    def _validate_type(obj: Any) -> bool:
+        if not type(obj) in (int, float, bool, str):
+            raise ValueError(
```

Comment:
should this throw or just return False? if we throw, won't the calling code stop execution of the for loop?

Reply:
Yes, the idea is to stop the execution if the user sends something we cannot analyze. I was trying to find a better way to do this (without the return statement) but couldn't find a good approach, since this is all iterators. Any suggestions?

Reply:
You don't need to return anything from this. Just do raise or return void.

Reply:
The function is called in a way which would only return values for which this returns True. I tried playing around with it but couldn't find a good solution. It's probably my lack of knowledge of Python generators TBH.

Reply:
Well, truthfully, looking at the code, you can just do range... 
something like:
validate(list_of_texts)
that will do range of the list_of_texts and won't return anything as the _validate_type is not really filtering anything and it's a little confusing to do it with:
list_of_texts = (text for text in list_of_texts if self._validate_type(text))
As it seems as though it is actually filtering something but it's not it's abruptly raise an exception

Reply:
`list_of_texts` is not really a list, it's an iterator.

Reply:
I'll think of other ways :)

Reply:
ohhh another option is to do it as part of going through the iterator in analyze_dict method which is perfectly fine to do validation on each and throwing an exception if needed.

Reply:
Fixed, moved to an external method which doesn't return True or does filtering